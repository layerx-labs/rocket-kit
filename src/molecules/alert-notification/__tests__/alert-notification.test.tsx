import React from 'react';
import { render, screen } from '@testing-library/react';
import 'jest-styled-components';
import { AlertNotification } from '../../..';

describe('AlertNotification', () => {
  it('renders', () => {
    const { asFragment } = render(
      <AlertNotification
        className="open"
        value="this is my notification text"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('has correct value', () => {
    const textValue = 'this is my notification text';
    render(<AlertNotification className="open" value={textValue} />);
    screen.queryByText(textValue);
  });
});
