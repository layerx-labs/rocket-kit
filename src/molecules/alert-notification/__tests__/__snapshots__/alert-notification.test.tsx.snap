// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertNotification renders 1`] = `
<DocumentFragment>
  .c1 {
  --button: hsl(186,62%,59%);
  --txt: hsl(0,0%,100%);
  --hover: #2ba8b6;
  --button: hsl(186,62%,59%);
  --hover: #2798a5;
  border-width: 0;
  border-style: solid;
  border-color: var(--button);
  border-radius: 0;
  background-color: transparent;
  width: 2.25rem;
  height: 2.25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0;
  text-transform: uppercase;
  white-space: nowrap;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  cursor: pointer;
}

.c1:hover {
  border-color: var(--hover);
  background-color: transparent;
}

.c1:hover span {
  color: var(--hover);
}

.c1:hover svg {
  fill: var(--hover);
}

.c1:disabled {
  cursor: inherit;
  opacity: 0.5;
}

.c1:disabled:hover {
  pointer-events: none;
}

.c1 > *:not(:last-child) {
  margin-left: 0;
  margin-right: 5px;
}

.c1 .spinner {
  border-top-color: var(--button);
}

.c1 span {
  position: relative;
  font-size: 0.75rem;
  font-weight: 700;
  color: var(--button);
  pointer-events: none;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

.c1 svg {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  width: auto;
  min-width: 1.25rem;
  height: 1.25rem;
  fill: var(--button);
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c0 {
  --shadow: 0 0 1.5625rem 0 rgba(40,40,40,0.2);
  position: fixed;
  top: 0.9375rem;
  right: 0;
  border: 2px solid #2ba8b6;
  border-radius: 6px;
  background-color: hsl(186,62%,59%);
  width: 18.75rem;
  max-height: -webkit-min-content;
  max-height: -moz-min-content;
  max-height: min-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.9375rem;
  color: hsl(0,0%,100%);
  -moz-box-shadow: var(--shadow);
  -webkit-box-shadow: var(--shadow);
  box-shadow: var(--shadow);
  z-index: 999;
  opacity: 0;
}

.c0 button {
  background-color: transparent;
}

.c0 button svg {
  fill: #2ba8b6;
}

.c0.open {
  -webkit-animation-name: slide-in;
  animation-name: slide-in;
  -webkit-animation-duration: 0.5s;
  animation-duration: 0.5s;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.c0 span {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-right: 0.3125rem;
  font-size: 0.85rem;
}

.c0 button {
  position: absolute;
  top: 0.25rem;
  right: 0;
  margin: 0 !important;
  width: 1.25rem;
  height: 1.25rem;
}

@media (min-width:1280px) {
  .c1 {
    min-width: 2.25rem;
  }
}

<div
    class="c0 open"
  >
    <div>
      <span>
        this is my notification text
      </span>
    </div>
    <button
      class="c1 button"
      color="primary"
      value=""
    >
      <svg
        class="icon"
        viewBox="0 0 32 32"
      >
        <path
          d="M24.4 7.613c-0.52-0.52-1.36-0.52-1.88 0l-6.52 6.507-6.52-6.52c-0.52-0.52-1.36-0.52-1.88 0s-0.52 1.36 0 1.88l6.52 6.52-6.52 6.52c-0.52 0.52-0.52 1.36 0 1.88s1.36 0.52 1.88 0l6.52-6.52 6.52 6.52c0.52 0.52 1.36 0.52 1.88 0s0.52-1.36 0-1.88l-6.52-6.52 6.52-6.52c0.507-0.507 0.507-1.36 0-1.867z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
